pkg/clientwrappers/dataflow/dataflow.go

```diff
@@ -359,6 +359,10 @@ func SyncDataflow(
 			return status, err
 		}
 		flow.Status = *status
+
+		if err := UnscheduleDataflow(flow, config); err != nil {
```

Comment:
May i ask why? What behavior are you seeing?

If the flow has a drain strategy, then the operator stops all input processors to let the rest of the connections drain. If it's stopped fully, the flow won't drain.

This might immediately be undone by the next `localChanged()` block bc it reverts any local changes.

Reply:
The operator will perform the drain/drop strategy as usual until the flowfile has no more flowfiles. Once the flow is empty the operator will stop it fully so the rollback or change version can be performed (in 1.16+). After the rollback or change version is done, the operator will reattached the parameter context, restart the services and the processors.

Reply:
Ah i see. The `prepareUpdatePG()` method will wait until the drop/drain is finished, by returning an error, before this new block is executed. Thank you for clarifying!